### ISSUE TITLE
redis connection error handling is erroneous ('sails is not defined')

### ISSUE THREAD

**Issue by naderio** [balderdashy/sails-redis#86]
```
/home/user/Workspace/project/project-api/node_modules/sails-redis/lib/connection.js:78
    sails && sails.log.error('RedisClient::Events[error], ' + err);
    ^

ReferenceError: sails is not defined
    at RedisClient.<anonymous> (/home/user/Workspace/project/project-api/node_modules/sails-redis/lib/connection.js:78:5)
    at emitOne (events.js:77:13)
    at RedisClient.emit (events.js:169:7)
    at RedisClient.on_error (/home/user/Workspace/project/project-api/node_modules/sails-redis/node_modules/redis/index.js:196:10)
    at Socket.<anonymous> (/home/user/Workspace/project/project-api/node_modules/sails-redis/node_modules/redis/index.js:106:14)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at emitErrorNT (net.js:1253:8)
    at doNTCallback2 (node.js:439:9)
    at process._tickCallback (node.js:353:17)
```
Error source is located in redis connection error handling at https://github.com/balderdashy/sails-redis/blob/fd57d66cd7624155d1eec286f32357b86ab3727e/lib/connection.js#L78

Variable `sails` is not defined and definition check is not  correct.

**Comment by catdad** [comment]
I'm running into the same issue. Since all that method does is log to a nonexistent variable (i.e. nothing), I am tempted to just remove it and submit a PR. However, I would really like to hear from the project maintainers about this issue. I am not comfortable shipping a product that will basically throw ever time there is a connection error (instead of calling callbacks with an error).

@shamasis Do you know anything about this? Committed here: https://github.com/balderdashy/sails-redis/commit/fd57d66cd7624155d1eec286f32357b86ab3727e

**Comment by catdad** [comment]
To add some details:

I am using `waterline` and `sails-redis` for a database outside of a sails application. I am not sure if that error logging does actually work in a sails application -- therefore, I don't want to just delete it outright.

Further, if I initialize a database (`waterline.initialize`) using `sails-redis` while redis is not running or cannot be connected to (bad config), I do not get a callback. This is bad. I believe that a callback is necessary to inform the user that a connection error occurred. Chances are that there is something wrong with my app and it should not continue running.

**Comment by Ryanc1256** [comment]
Hey guys, I fixed the log issue, it should revert to the console if there is sails isn't defined now nice spotting... also if you want a callback, please make another issue too... 


I'm just thinking what the other database adapters do with this problem anywho.

**Comment by shamasis** [comment]
Hmm. Guess if it's in other adapters, those will need changes too.

**Comment by catdad** [comment]
I checked postgresql and mongo adapters, and they both return an error. I didn't check any others, but it seems good practice. I have submitted https://github.com/balderdashy/sails-redis/issues/87

**Comment by catdad** [comment]
Also, what is the usual turnaround before a fix makes it into NPM? Would it be reasonable to assume sometime this week?

**Comment by Ryanc1256** [comment]
ill get the guys to push it to npm since i cant do it my self

**Comment by particlebanana** [comment]
@Ryanc1256 @catdad on it

**Comment by particlebanana** [comment]
alright published `0.10.7`

**Comment by shamasis** [comment]
Thanks @particlebanana!
